Client/Frontend/Login/LoginListViewController.swift

```diff
@@ -0,0 +1,352 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+import Shared
+import BraveShared
+import Storage
+import Data
+
+private let log = Logger.browserLogger
+
+class LoginListViewController: LoginAuthViewController {
+    
+    // MARK: UX
+    
+    struct UX {
+        static let headerHeight: CGFloat = 44
+    }
+    
+    // MARK: Constants
+    
+    struct Constants {
+        static let saveLoginsRowIdentifier = "saveLoginsRowIdentifier"
+    }
+    
+    // MARK: Section
+    
+    enum Section: Int, CaseIterable {
+        case options
+        case savedLogins
+    }
+    
+    weak var settingsDelegate: SettingsDelegate?
+    
+    // MARK: Private
+    
+    private let profile: Profile
+    private var loginEntries = [Login]()
+    private var isFetchingLoginEntries: Bool = false
+    private var searchLoginTimer: Timer?
+
+    private let searchController = UISearchController(searchResultsController: nil)
+    private let emptyLoginView = LoginEmptyView()
+    
+    // MARK: Lifecycle
+    
+    init(profile: Profile) {
+        self.profile = profile
+        super.init(requiresAuthentication: true)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        applyTheme()
+        
+        // Insert Done button if being presented outside of the Settings Navigation stack
+        if navigationController?.viewControllers.first === self {
+            navigationItem.leftBarButtonItem =
+                UIBarButtonItem(title: Strings.settingsSearchDoneButton, style: .done, target: self, action: #selector(dismissAnimated))
+        }
+
+        navigationItem.do {
+            $0.searchController = searchController
+            $0.hidesSearchBarWhenScrolling = false
+            $0.rightBarButtonItem = editButtonItem
+            $0.rightBarButtonItem?.isEnabled = !self.loginEntries.isEmpty
+        }
+        definesPresentationContext = true
+
+        tableView.tableFooterView = SettingsTableSectionHeaderFooterView(
+            frame: CGRect(width: tableView.bounds.width, height: UX.headerHeight))
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        fetchLoginInfo()
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        
+        tableView.endEditing(true)
+    }
+    
+    // MARK: Internal
+    
+    private func applyTheme() {
+        navigationItem.title = Strings.Login.loginListNavigationTitle
+
+        tableView.do {
+            $0.accessibilityIdentifier = "Logins Passwords List"
```

Comment:
can be translated

Reply:
Fix at https://github.com/brave/brave-ios/pull/4124/commits/ded5065d24f539a7dee6bb8bcb40b93a2b70fe7b